sonar-plugin-api/src/main/java/org/sonar/api/server/rule/ContextAwareRuleDescriptionSection.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Sonar Plugin API
+ * Copyright (C) 2009-2022 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.api.server.rule;
+
+import java.util.Optional;
+
+import static java.util.Objects.requireNonNull;
+
+class ContextAwareRuleDescriptionSection extends DefaultRuleDescriptionSection implements RuleDescriptionSection {
+
+  private final Context context;
+
+  ContextAwareRuleDescriptionSection(String key, Context context, String htmlContent) {
```

Comment:
nitpick: Context could be third argument

Reply:
It was actually like that at first, and I changed because to me it seemed a bit odd. As it is currently, the 2 first parameters are the components of the key. The 3rd is defined for a specific combination of both key + context.